tests/common/include/tests/common/tools/intersectors/helix_intersector.hpp

```diff
@@ -0,0 +1,23 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+namespace detray {
+
+/// @brief Intersection implementation for detector surfaces using a helix
+/// trajectory.
+///
+/// The algorithm uses the Newton-Raphson method to find an intersection on
+/// the unbounded surface and then applies the mask.
+///
+/// @note specialized into @c helix_plane_intersector and
+/// @c helix_cylinder_intersector
+template <typename transform3_t, typename mask_t, typename = void>
+struct helix_intersector {};
```

Comment:
It makes the call in the ```helix_intersection_kernel``` a bit nicer with less code duplication (no ```if constexpr ... ```)
The reason it does not exist in the 'production' intersectors is that there the mask shapes provide the correct intersector, until I figure out if I need special intersectors for SoA layout or not. For the helix intersector, the situation is inverted, because I don't want to redefine the mask shape types and use two different toy detectors in the navigation validation. Here, the mask type is used to infer whether we need to call the plane or the cylinder helix intersector and that is done via this class and template specialization now

Reply:
Hmm I wonder this kind of interface is possible

```c++
const auto ray_result = mask.intersector<ray_t>()(traj,surface);
const auto hlx_result = mask.intersector<helix_t>()(traj,surface);
```

Reply:
Right, but I am not sure how to do it without coupling the particle intersection code to the navigation intersection code in the core, which would also be a bad design in my opinion. I rather have ugly interfaces in the particle gun for now :)

Reply:
I don't know how there could be a coupling between them - but let's deal with that later.

Reply:
I meant, by putting the helix intersectors as overloads into the core intersectors. The helix intersectors are primarily written for the particle gun

Reply:
I don't think we don't have to put the helix intersectors into core intersector as long as we keep it as templated function.
```c++
class rectangle{
   template < typename trajectory_t>
   using intersector_type = plane_intersector<trajectory_t>

   template < typename trajectory_t >
   intersector_type<trajectory_t> intersector(trajectory_t traj, surface){
       return intersector_type<trajectory_t>();
   }
}; 
```